Rename rendering components for improved consistency and clarity

Jondolf opened the pull request:
# Objective

The names of numerous rendering components in Bevy are inconsistent and a bit confusing. Relevant names include:

- `AutoExposureSettings`
- `AutoExposureSettingsUniform`
- `BloomSettings`
- `BloomUniform` (no `Settings`)
- `BloomPrefilterSettings`
- `ChromaticAberration` (no `Settings`)
- `ContrastAdaptiveSharpeningSettings`
- `DepthOfFieldSettings`
- `DepthOfFieldUniform` (no `Settings`)
- `FogSettings`
- `SmaaSettings`, `Fxaa`, `TemporalAntiAliasSettings` (really inconsistent??)
- `ScreenSpaceAmbientOcclusionSettings`
- `ScreenSpaceReflectionsSettings`
- `VolumetricFogSettings`

Firstly, there's a lot of inconsistency between `Foo`/`FooSettings` and `FooUniform`/`FooSettingsUniform` and whether names are abbreviated or not.

Secondly, the `Settings` post-fix seems unnecessary and a bit confusing semantically, since it makes it seem like the component is mostly just auxiliary configuration instead of the core *thing* that actually enables the feature. This will be an even bigger problem once bundles like `TemporalAntiAliasBundle` are deprecated in favor of required components, as users will expect a component named `TemporalAntiAlias` (or similar), not `TemporalAntiAliasSettings`.

## Solution

Drop the `Settings` post-fix from the component names, and change some names to be more consistent.

- `AutoExposure`
- `AutoExposureUniform`
- `Bloom`
- `BloomUniform`
- `BloomPrefilter`
- `ChromaticAberration`
- `ContrastAdaptiveSharpening`
- `DepthOfField`
- `DepthOfFieldUniform`
- `DistanceFog`
- `Smaa`, `Fxaa`, `TemporalAntiAliasing` (note: we might want to change to `Taa`, see "Discussion")
- `ScreenSpaceAmbientOcclusion`
- `ScreenSpaceReflections`
- `VolumetricFog`

I kept the old names as deprecated type aliases to make migration a bit less painful for users. We should remove them after the next release. (And let me know if I should just... not add them at all)

I also added some very basic docs for a few types where they were missing, like on `Fxaa` and `DepthOfField`.

## Discussion

- `TemporalAntiAliasing` is still inconsistent with `Smaa` and `Fxaa`. Consensus [on Discord](https://discord.com/channels/691052431525675048/743663924229963868/1280601167209955431) seemed to be that renaming to `Taa` would probably be fine, but I think it's a bit more controversial, and it would've required renaming a lot of related types like `TemporalAntiAliasNode`, `TemporalAntiAliasBundle`, and `TemporalAntiAliasPlugin`, so I think it's better to leave to a follow-up.
- I think `Fog` should probably have a more specific name like `DistanceFog` considering it seems to be distinct from `VolumetricFog`. ~~This should probably be done in a follow-up though, so I just removed the `Settings` post-fix for now.~~ (done)

---

## Migration Guide

Many rendering components have been renamed for improved consistency and clarity.

- `AutoExposureSettings` → `AutoExposure`
- `BloomSettings` → `Bloom`
- `BloomPrefilterSettings` → `BloomPrefilter`
- `ContrastAdaptiveSharpeningSettings` → `ContrastAdaptiveSharpening`
- `DepthOfFieldSettings` → `DepthOfField`
- `FogSettings` → `DistanceFog`
- `SmaaSettings` → `Smaa`
- `TemporalAntiAliasSettings` → `TemporalAntiAliasing`
- `ScreenSpaceAmbientOcclusionSettings` → `ScreenSpaceAmbientOcclusion`
- `ScreenSpaceReflectionsSettings` → `ScreenSpaceReflections`
- `VolumetricFogSettings` → `VolumetricFog`

alice-i-cecile commented:
> I think Fog should probably have a more specific name like DistanceFog considering it seems to be distinct from VolumetricFog. This should probably be done in a follow-up though, so I just removed the Settings post-fix for now.

Agreed!

JMS55 commented:
Imo we should do the fog -> distance fog in this PR.

I'd also like if you could rename variables and bundle fields to remove "settings".

Jondolf commented:
> Imo we should do the fog -> distance fog in this PR.

Done!

> I'd also like if you could rename variables and bundle fields to remove "settings".

This feels like it's a bit out of scope for this PR, but I renamed a bunch of variables to remove `_settings` and overall tried to make names clearer where it made sense. Also added some basic docs that were missing.

I don't think we want to do this for bundles though. The impression I got is that we don't want to cause any unnecessary breaking changes for them (like remove or rename properties), considering the plan is to deprecate (most of) the first-party bundles in 0.15.

Jondolf commented:
> Done!

Hmm or I guess for fog I should also rename `GpuFog`, `FogMeta`, and maybe some shader stuff to have the `Distance` prefix as well? I'm not familiar with the fog rendering at all, so I don't really know if any of it is shared between the different fog types. To me it looks like basically all of the general `Fog` and `GpuFog` stuff is only for distance fog and atmospheric fog, but I'm not 100% sure, which is a part of why I would've liked to leave that stuff to a follow-up / someone who knows what all the types actually do 😅

alice-i-cecile commented:
Agreed on leaving the bundles alone. I'm fine to leave the fog stuff alone at this point: I like the initial rename to avoid a double-rename, making it harder to compile the migration guide.

MrGVSV commented:
> TemporalAntiAliasing is still inconsistent with Smaa and Fxaa. Consensus on Discord seemed to be that renaming to Taa would probably be fine, but I think it's a bit more controversial, and it would've required renaming a lot of related types like TemporalAntiAliasNode, TemporalAntiAliasBundle, and TemporalAntiAliasPlugin, so I think it's better to leave to a follow-up.

I have no knowledge on rendering or what any of this stuff does haha but what about expanding `Smaa` and `Fxaa`?

#If I know I need "anti-aliasing", I imagine it would be really helpful to just type that in on the docs to find all my options. You could probably achieve that with a doc alias though but I don't think that helps when doing the same sort of thing in my IDE and letting intellisense be my quasi-doc-search.

Although, I imagine `SubpixelMorphologicalAntiAliasingNeighborhoodBlendingPipelineKey` is the friendliest of identifiers lol.

You could maybe do it for the "main component" but I guess that could be seen as adding more inconsistencies.

In either case (not necessarily part of this PR), the docs for these acronyms should at least start out with what they stand for. `SmaaSettings` puts it at the end of the sentence and `Fxaa` doesn't seem to have any.

Jondolf commented:
> In either case (not necessarily part of this PR), the docs for these acronyms should at least start out with what they stand for. SmaaSettings puts it at the end of the sentence and Fxaa doesn't seem to have any.

I actually added basic docs for these in this PR already, along with adding the long names as doc aliases (even though it's technically not related to this PR). See [`Fxaa`](https://github.com/bevyengine/bevy/pull/15035/files#diff-6155eb8e58c32ac0e2e22c5d77199a90f5993dd413118e6ce178e05de5db33e8) and [`Smaa`](https://github.com/bevyengine/bevy/pull/15035/files#diff-f0a339d5d21b91db295f544748e6cfdf1988ecd304cf6cf6bbaaac5e1ac1f7a5) in the diff :)

Jondolf commented on crates/bevy_core_pipeline/src/smaa/mod.rs:
```diff
@@ -95,17 +95,21 @@ const SMAA_SEARCH_LUT_TEXTURE_HANDLE: Handle<Image> = Handle::weak_from_u128(318
 #[derive(Clone, Copy, Default, Component, Reflect, ExtractComponent)]
 #[reflect(Component, Default)]
-pub struct SmaaSettings {
+#[doc(alias = "SubpixelMorphologicalAntialiasing")]
```
Whoops, missed capitalization here
```suggestion
#[doc(alias = "SubpixelMorphologicalAntiAliasing")]
```

cart reviewed (approved):
Much nicer!
